### PR TITLE
include player rank in profile page

### PR DESCRIPTION
**Pull request by lucasgautheron**
No idea what i was doing, so thats probably bad code-wise - but it works

### CODE REVIEW

File: web/app/controllers/PlayersController.scala [hunk 1]
```diff
@@ -43,10 +43,17 @@ class PlayersController @Inject()(common: Common, referenceProvider: ReferencePr
 
   def player(id: String) = Action.async { implicit request =>
     async {
+      import _root_.players.PlayerStat
+      import _root_.players.PlayersStat
+      import _root_.players.PlayersStat.ImplicitWrites._
+      val defaultStats = PlayersStat.empty("", "", "")
```
Comment (ScalaWilliam) [inline web/app/controllers/PlayersController.scala]: Don't do 'default' stats if there are no stats.

File: web/app/controllers/PlayersController.scala [hunk 2]
```diff
@@ -43,10 +43,17 @@ class PlayersController @Inject()(common: Common, referenceProvider: ReferencePr
...
+      val stats = await(fullProvider.playerRanks).onlyRanked.players.getOrElse(id, defaultStats)
       await(fullProvider.getPlayerProfileFor(id)) match {
         case Some(player) =>
           await(renderJsonWR("/player.php")(_.withQueryString("id" -> id))(
-            Map("player" -> player.toJson)
+            Map("player" -> player.toJson,
+                "stats"  -> Json.toJson(stats)
+            )
```
Comment (ScalaWilliam) [inline web/app/controllers/PlayersController.scala]: if stats = await(...).onlyRanked.players.get(id)

then you can do: val jsonMap = stats match { case Some(actualStats) => Map("player"->..., "stats" -> Json.toJson(actualStats)); case None => Map("player"->,...)}

File: web/dist/www/player.php [hunk 3]
```diff
@@ -149,6 +150,14 @@ function none_achievement($achievement)
                         <th>Games played</th><td><?php echo $user['stats']['gamesPlayed']; ?></td>
                         <th>Frags</th><td><?php echo $user['stats']['frags']; ?></td>
                     </tr>
+                    
+                    <?php if($stats['user']) { ?>
```
Comment (ScalaWilliam) [inline web/dist/www/player.php]: hmm now we have a duplication of stuff. E.g. $user['stats'] and $stats. Confusing.

How about naming it ranks?

Comment (lucasgautheron) [inline web/dist/www/player.php]: yep i agree, i didn't like it myself

File: web/app/controllers/PlayersController.scala (hunk 2 above)
Comment (lucasgautheron) [inline web/app/controllers/PlayersController.scala]: Alright, thanks, will change that then

**Comment by lucasgautheron** [comment]
PR build failing because of : 
```
[info] - should Fire off new games *** FAILED ***

[info]   0 did not equal 1 (JournalNewGamesTest.scala:63)
```

Which is independent from the PR itself, tested locally and working

File: accumulation/src/main/scala/af/FullIterator.scala [hunk 4]
```diff
@@ -112,6 +116,12 @@ case class FullProfile(user: User, recentGames: List[JsonGame], achievements: Op
       ))
       jsObject = jsObject.deepMerge(no)
     }
+    
+    rank match {
```
Comment (ScalaWilliam) [inline accumulation/src/main/scala/af/FullIterator.scala]: this doesn't actually do anything?

File: accumulation/src/main/scala/af/FullIterator.scala [hunk 5]
```diff
@@ -93,16 +95,18 @@ case class FullIterator
...
-      FullProfile(user, recentGames, achievements)
+      val rank = playersStats.onlyRanked.players.get(id)
+      FullProfile(user, recentGames, achievements, rank)
     }
 
 }
 
-case class FullProfile(user: User, recentGames: List[JsonGame], achievements: Option[PlayerState]) {
+case class FullProfile(user: User, recentGames: List[JsonGame], achievements: Option[PlayerState], rank: Option[PlayerStat]) {
   def toJson = {
```
Comment (ScalaWilliam) [inline accumulation/src/main/scala/af/FullIterator.scala]: Also I mentioned this method is deprecated, has to be deleted.

**Comment by ScalaWilliam** [comment]
When you fix that merge

File: web/app/views/player/player.scala.html [hunk 6]
```diff
@@ -16,6 +16,20 @@
...
+                    @fullProfile.rank match {
+                        case Some(rank) => {
+                            <tr>
+                                <th>Elo Rank</th> <td>@rank.rank</td>
+                                <th>Elo Points</th> <td>@Math.round(rank.elo)</td>
+                            </tr>
+                        }
+                        case _ => {
+                            <tr>
+                                <th>Elo Rank</th> <td>Play more to get ranked</td>
```
Comment (ScalaWilliam) [inline web/app/views/player/player.scala.html]: "Play more"

File: accumulation/src/main/scala/af/FullIterator.scala [hunk 7]
```diff
@@ -93,16 +95,18 @@ case class FullIterator
         .collect { case (_, game) if game.hasUser(user.id) => game }
         .toList.sortBy(_.id).takeRight(7).reverse
       val achievements = achievementsIterator.map.get(id)
-      FullProfile(user, recentGames, achievements)
+      val rank = playersStats.onlyRanked.players.get(id)
+      FullProfile(user, recentGames, achievements, rank)
```
Comment (ScalaWilliam) [inline accumulation/src/main/scala/af/FullIterator.scala]: Maybe could use named params now? It's getting  quite a few arguments now

File: accumulation/src/main/scala/af/FullIterator.scala [hunk 8]
```diff
@@ -4,6 +4,8 @@ import acleague.enrichers.JsonGame
 import acleague.ranker.achievements.immutable.PlayerStatistics
 import acleague.ranker.achievements.{Jsons, PlayerState}
 import clans.{CompleteClanwar, Clanstats, Clanwars}
+import players.PlayerStat
+import players.PlayersStat.ImplicitWrites._
```
Comment (ScalaWilliam) [inline accumulation/src/main/scala/af/FullIterator.scala]: You won't need this import any more after removing toJson

**Comment by ScalaWilliam** [comment]
Great :+1:

**Comment by lucasgautheron** [comment]
:+1:

**Pull request merged**